### DISCUSSION TITLE
Arcus.Templates.AzureFunctions.Http keyvault implementation

### DISCUSSION THREAD

**Discussion by pim-simons** [arcus-azure/arcus.templates#317] [Q&A] [answered]
Working with the Arcus.Templates.AzureFunctions.Http, after installation it uses:
`stores.AddAzureKeyVaultWithManagedServiceIdentity("https://your-keyvault.vault.azure.net/");`
Which gives the warning: 
`SecretStoreBuilderExtensions.AddAzureKeyVaultWithManagedServiceIdentity(SecretStoreBuilder, string, string, string, bool)' is obsolete: 'Use the AddAzureKeyVaultWithManagedIdentity overload with the 'clientId' instead`

I managed to get it all working with this code:
 ```
var keyVaultUri = config.GetValue<string>("KEYVAULT_URI");
stores.AddAzureKeyVaultWithManagedIdentity(keyVaultUri, allowCaching: true);
 ```
But this still gives me the warning:
`SecretStoreBuilderExtensions.AddAzureKeyVaultWithManagedIdentity(SecretStoreBuilder, string, bool)' is obsolete: 'Use the AddAzureKeyVaultWithManagedIdentity method overload with Default instead to allow default caching`

What is the actual best practice way (no warnings) of implementing this?

Apologies if this should be posted in Arcus.Security, but since it is used in the Arcus.Templates.AzureFunctions.Http template I figured to post it here.

**Comment by tomkerkhove** [comment, marked as answer]
We'll update them, sorry about that! https://github.com/arcus-azure/arcus.templates/issues/318

You can find the latest docs on https://security.arcus-azure.net/features/secret-store/provider/key-vault
